src/hints/state.rs

```diff
@@ -323,6 +326,48 @@ pub fn enter_scope_commitment_info_by_address(
     Ok(())
 }
 
+pub const WRITE_SPLIT_RESULT: &str = indoc! {r#"
+    from starkware.starknet.core.os.data_availability.bls_utils import split
+
+    segments.write_arg(ids.res.address_, split(ids.value))"#
+};
+pub fn write_split_result(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    // this hint fills in a Cairo BigInt3 by taking a felt (ids.value) and passing it to a split fn
+    let value = get_integer_from_var_name(vars::ids::VALUE, vm, ids_data, ap_tracking)?;
+    let res_ptr = get_ptr_from_var_name(vars::ids::RES, vm, ids_data, ap_tracking)?;
```

Comment:
I interpret `.address_` as getting a pointer from a variable, so IMO you should use `get_relocatable_from_var_name`.

Reply:
Oh, this clears up a misconception I had. Most examples we have that use `address_` use `get_ptr_from_var_name()`, but they also are dereferencing an outer struct in those cases.

Reply:
213cf16